### PR TITLE
fix: v1 to v2 migration, update labels put by control-plane

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/agents/src/bin/core/controller/resources/migration.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    let prefix = &product_v1_key_prefix_obj(spec_type);
+    let store_entries = store.get_values_prefix(prefix).await?;
+    for (k, v) in &store_entries {
+        let id = k
+            .split('/')
+            .last()
+            .ok_or_else(|| StoreError::InvalidKey { key: k.to_string() })?;
+
+        let new_value = match spec_type {
+            StorableObjectType::VolumeSpec => migrate_volume_labels(v.clone())
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/resources/migration.rs]: can we consume store_entries and skip clone here?

File: control-plane/agents/src/bin/core/controller/resources/migration.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    types::v0::{
+        store::{definitions::StoreError, pool::PoolSpec, volume::VolumeSpec},
+        transport::PoolTopology,
+    },
+};
+use tracing::{info, warn};
+use utils::{CREATED_BY_KEY, DSP_OPERATOR, MSP_OPERATOR};
+
+/// Migrate the values from legacy keys to latest keys.
+pub async fn migrate_product_v1_to_v2<S: Store>(
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/resources/migration.rs]: pub(crate)

File: control-plane/agents/src/bin/core/controller/resources/migration.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    store.delete_values_prefix(prefix).await?;
+    warn!("Removed {spec_type:?} for v1 key space from store");
+
+    Ok(())
+}
+
+/// Migrate the labels put by product v1 csi-controller to latest.
+pub fn migrate_volume_labels(v: Value) -> Result<Value, serde_json::Error> {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/resources/migration.rs]: pub(crate)

File: control-plane/agents/src/bin/core/controller/resources/migration.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    }
+                }
+            }
+        }
+    }
+    serde_json::to_value(&spec)
+}
+
+/// Migrate the labels put by product v1 operator to latest.
+pub fn migrate_pool_labels(v: Value) -> Result<Value, serde_json::Error> {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/controller/resources/migration.rs]: pub(crate)

File: utils/pstor/src/error.rs [hunk 5]
```diff
@@ -59,6 +59,9 @@ pub enum Error {
     /// Empty value.
     #[snafu(display("Failed to get value as string. Error {}", source))]
     ValueString { source: etcd_client::Error },
+    /// Serde error.
+    #[snafu(display("Serde error. Error {}", source))]
+    SerdeError { source: serde_json::Error },
```
Comment (tiagolobocastro) [inline utils/pstor/src/error.rs]: Can't we use simply change the migrate methods to return this  error with the appropriate already defined SerialiseValue/DeserialiseValue errors?

File: utils/utils-lib/src/constants.rs [hunk 6]
```diff
@@ -45,6 +45,9 @@ pub const CREATED_BY_KEY: &str = "openebs.io/created-by";
 /// The value to mark the creation source of a pool to be disk pool operator in labels.
 pub const DSP_OPERATOR: &str = "operator-diskpool";
 
+/// The value to mark the creation source of a pool to be msp operator in labels.
+pub const MSP_OPERATOR: &str = "msp-operator";
```
Comment (tiagolobocastro) [inline utils/utils-lib/src/constants.rs]: maybe keep this in the migration module, since we don't need it otherwise.

**Review by tiagolobocastro — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

File: utils/pstor/src/error.rs (hunk 5 above)
Comment (Abhinandan-Purkait) [reply utils/pstor/src/error.rs]: Yeah, changed to that, removed this.

File: control-plane/agents/src/bin/core/controller/resources/migration.rs (hunk 2 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/resources/migration.rs]: Done.

File: control-plane/agents/src/bin/core/controller/resources/migration.rs (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/resources/migration.rs]: Done.

File: control-plane/agents/src/bin/core/controller/resources/migration.rs (hunk 3 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/resources/migration.rs]: Doesn't even need to be pub(crate) changed to private.

File: control-plane/agents/src/bin/core/controller/resources/migration.rs (hunk 4 above)
Comment (Abhinandan-Purkait) [reply control-plane/agents/src/bin/core/controller/resources/migration.rs]: Doesn't even need to be pub(crate) changed to private.

File: utils/utils-lib/src/constants.rs (hunk 6 above)
Comment (Abhinandan-Purkait) [reply utils/utils-lib/src/constants.rs]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**